File: backend/src/main/java/edu/jmi/openatom/server/openatomsystem/entity/MemberProfile.java
package edu.jmi.openatom.server.openatomsystem.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.math.BigDecimal;
import java.sql.Timestamp;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("member_profile")
public class MemberProfile {
  @TableId(value = "id", type = IdType.AUTO)
  private Long id;

  @TableField("user_id")
  private Integer userId;

  private String slug;

  @TableField("display_name")
  private String displayName;

  private String headline;
  private String bio;

  @TableField("avatar_url")
  private String avatarUrl;

  @TableField("banner_url")
  private String bannerUrl;

  @TableField("card_background_url")
  private String cardBackgroundUrl;

  @TableField("card_focus_x")
  private BigDecimal cardFocusX;

  @TableField("card_focus_y")
  private BigDecimal cardFocusY;

  @TableField("theme_key")
  private String themeKey;

  @TableField("color_mode")
  private String colorMode;

  private String visibility;
  private String status;

  @TableField("comments_enabled")
  private Boolean commentsEnabled;

  @TableField("show_department")
  private Boolean showDepartment;

  @TableField("show_position")
  private Boolean showPosition;

  private String skills;
  private Integer version;

  @TableField("published_at")
  private Timestamp publishedAt;

  @TableField("created_at")
  private Timestamp createdAt;

  @TableField("updated_at")
  private Timestamp updatedAt;
}

File: frontend/web_pc/src/types/member-profile.ts
export interface MemberCard {
  slug: string
  displayName: string
  headline?: string
  avatarUrl?: string
  cardBackgroundUrl?: string
  cardFocusX?: number
  cardFocusY?: number
  departmentName?: string
  positionName?: string
  skills: string[]
  articleCount: number
  likeCount: number
  liked: boolean
  customized: boolean
}

export interface ProfileArticle {
  id: number
  title: string
  summary?: string
  coverUrl?: string
  publishedAt?: string
}

export interface ProfileModule {
  id?: number
  key: string
  title?: string
  sortOrder: number
  columnSpan: number
  enabled: boolean
  visibility: 'members' | 'private'
  data: Record<string, any>
}

export interface ProfileSocialLink {
  id?: number
  platform: string
  label?: string
  url: string
  sortOrder: number
  enabled: boolean
}

export interface MemberProfile {
  slug: string
  displayName: string
  headline?: string
  bio?: string
  avatarUrl?: string
  bannerUrl?: string
  cardBackgroundUrl?: string
  cardFocusX: number
  cardFocusY: number
  departmentName?: string
  positionName?: string
  skills: string[]
  themeKey: 'minimal' | 'tech' | 'warm' | 'editorial'
  colorMode: 'light' | 'dark' | 'system'
  visibility: 'members' | 'unlisted' | 'private'
  status: 'default' | 'draft' | 'published'
  commentsEnabled: boolean
  showDepartment: boolean
  showPosition: boolean
  version: number
  owner: boolean
  customized: boolean
  likeCount: number
  liked: boolean
  commentCount: number
  publishedAt?: string
  updatedAt?: string
  modules: ProfileModule[]
  socialLinks: ProfileSocialLink[]
}

export interface MemberPage {
  list: MemberCard[]
  page: number
  pageSize: number
  total: number
}

export interface MemberFilters {
  departments: Array<{ id: number; name: string }>
  skills: string[]
}
